### DISCUSSION TITLE
RTSP Camera Additions

### DISCUSSION THREAD

**Discussion by 400HPMustang** [OctoFarm/OctoFarm#512] [Feature Request]
Just wondering if it's possible to implement a way to add an RTSP stream from something like a Wyze cam instead of a webcam stream?

**Comment by NotExpectedYet** [comment]
> Just wondering if it's possible to implement a way to add an RTSP stream from something like a Wyze cam instead of a webcam stream?

It's a feature were looking to implement but it requires the stream to be transcoded if we did so I need to do some testing to see how a Pi3+, which we support, would handle the streams + OctoFarm as well.  

For a work around for now there's two options. 

1. Run a middle man software such as motionEye on another device. This takes in a RTSP:// URL and converts it to mjpeg which octoprint and OctoFarm support. 

2. Or do the transcoding on OctoPrint with ffmpeg. The guide I posted below should help. You need wyze V2 I believe which has the RTSP support and then ffmpeg installed locally on that device. 
https://medium.com/@adamargo/using-a-wyze-cam-with-octoprint-on-a-raspberry-pi-7a1c886b9afe
